### PR TITLE
Fixes #135 - Fix rendering of buttons in header

### LINKED ISSUE #135: Buttons are not parsed in templates (theme)
{$isango_buttons},{$isango_input['min']}, {$isango_input['max']}, {$isango_input['pop']}

Are not parsed... only works in UCP.

@effone

### PR DESCRIPTION
**Pull request by Eldenroot**
Use global_start instead of global_end  hook to render buttons in header.

Fixes #135 

**Thank you @lairdshaw for debugging and for your fix!**

### CODE REVIEW

**Comment by effone** [comment]
As much I can remember I used global start only for templates as well as buttons...
I am so lost.

**Comment by lairdshaw** [comment]
> As much I can remember I used global start only for templates as well as buttons...

Looks like you switched to global end for buttons 17 days ago in commit 4106193fc3a1643a10234f11c5d7944f36e88bbc "Input loader hook change". Not sure what motivated that change.

> I am so lost.

Oh, but you're on GitHub. You're home!

**Review by lairdshaw — DISMISSED** [review]

**Comment by effone** [comment]
Can't remember. Probably to cover the template caching...

We need to check further if the button templates are cached now and if not need to finetune the task sequence / priority as both templates and buttons are using same hook.

We can alternately think of triggering a single hook for template caching and call the button generation from inside template cache function after caching templates just to make sure about the process sequence.

**Comment by lairdshaw** [comment]
Probably your suggestion is the way to go. Maybe have a single `global_start` hook called something like `isango_init()`, and from it call `isango_templates()` followed by `isango_buttons()`?

**Comment by lairdshaw** [comment]
Ah, wait. That wouldn't ensure that templates are cached - they need to be cached in-between the calls. Now I see what probably motivated you to change the hook.

**Comment by lairdshaw** [comment]
OK, it looks like instead of moving `isango_buttons` to the `global_start` hook, it would work if we moved it to the `global_intermediate` hook, which is called on line 494. That way it occurs after templates are cached on line 476,
but before the header is generated on line 958.

Yes?

**Comment by effone** [comment]
> Yes?

There is a hook like that? Yes thats the good approach.

The nasty one would be call `isango_buttons()` from inside `isango_templates()` at the end, after caching templates.

**Comment by effone** [comment]
Also to mention, the third parameter of `add_hook()` method is **priority**. 
Defining priority is another good approach too...

https://github.com/mybb/mybb/blob/feature/inc/class_plugins.php#L53

**Comment by effone** [comment]
> Better is to add isango_buttons() to the global_intermediate hook than to the global_start hook.

Agreed.

**Comment by lairdshaw** [comment]
> There is a hook like that? Yes thats the good approach.

I'm glad we're agreed. @Eldenroot, could you please update the PR so that we use the `global_intermediate` hook rather than `global_start`?

> The nasty one would be call `isango_buttons()` from inside `isango_templates()` at the end, after caching templates.

Mmm, but templates are only cached _after_ the `global_start` hook, on line 476 of global.php, so they aren't cached until _after_ `isango_templates()` runs.

> Defining priority is another good approach too...

Same problem as above: the templates are only cached _after_ `global_start`, so it wouldn't matter what priority the `global_start` hooks ran at - templates wouldn't be cached anyway.

File: upload/inc/plugins/isango.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 }
 
 $plugins->add_hook('global_start', 'isango_templates');
-$plugins->add_hook('global_end', 'isango_buttons');
+$plugins->add_hook('global_start', 'isango_buttons');
```
Comment (effone) [inline upload/inc/plugins/isango.php]: Please use `global_intermediate` as rightly suggested by @lairdshaw

**Review by effone — CHANGES_REQUESTED** [review]

**Comment by Eldenroot** [comment]
Ok, I will edit it in 2 or 3 hours.

**Comment by Eldenroot** [comment]
OK, changed. Can we merge?

**Review by effone — APPROVED** [review]

**Pull request merged**